pkg/controller.v1/tensorflow/controller.go

```diff
@@ -310,7 +310,9 @@ func (tc *TFController) syncTFJob(key string) (bool, error) {
 	}
 
 	tfjob := sharedTFJob.DeepCopy()
-	tfjobNeedsSync := tc.satisfiedExpectations(tfjob)
+
+	// Sync tfjob every time if EnableDynamicWorker is true
+	tfjobNeedsSync := tfjob.Spec.EnableDynamicWorker || tc.satisfiedExpectations(tfjob)
```

Comment:
Why do we need to sync every time?

Reply:
Let's say we have a 3-worker tfjob while one worker is pending for whatever reason (e.g, short of resource). Now we want to scale up/down the number of workers. If we still allow the expectation mechanism here, the scaling logic won't be triggered until the pending pod, which makes expectation unsatisfied, starts running.